spec/UseCaseValidationSpec.php

```diff
@@ -0,0 +1,117 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Marcosh\LamPHPda\ValidationSpec;
+
+use Marcosh\LamPHPda\Instances\ListL\ConcatenationMonoid;
+use Marcosh\LamPHPda\Validation\Validation;
+use Marcosh\LamPHPda\Validation\Validation as V;
+
+final class TinValidator {
+    public static function isFollowBelgiumRule1(string $tin): bool
+    {
+        $divisionRemainderBy97 = (int) (substr($tin, 0, 9)) % 97;
+
+        return 97 - $divisionRemainderBy97 === (int) (substr($tin, 9, 3));
+    }
+
+    public static function isFollowBelgiumRule2(string $tin): bool
+    {
+        $divisionRemainderBy97 = (int) ('2' . substr($tin, 0, 9)) % 97;
+
+        return 97 - $divisionRemainderBy97 === (int) (substr($tin, 9, 3));
+    }
+
+    public static function getDateType(string $tin): int
+    {
+        $year = (int) (substr($tin, 0, 2));
+        $month = (int) (substr($tin, 2, 2));
+        $day = (int) (substr($tin, 4, 2));
+
+        $y1 = checkdate($month, $day, 1900 + $year);
+        $y2 = checkdate($month, $day, 2000 + $year);
+
+        if (0 === $day || 0 === $month || ($y1 && $y2)) {
+            return 3;
+        }
+
+        if ($y1) {
+            return 1;
+        }
+
+        if ($y2) {
+            return 2;
+        }
+
+        return 0;
+    }
+
+    public static function hasValidDateType(string $tin): bool
+    {
+        return 0 !== self::getDateType($tin);
+    }
+
+    public static function followRule1(string $tin): bool
+    {
+        $dateType = self::getDateType($tin);
```

Comment:
instead of recomputing the date type every time you need it, it would be nice to store it in a temporary object

Reply:
This is something I wanted to do indeed, but couldn't find a proper way to do it here. Help is welcome.

Reply:
Oooh you're using a static variable, ok I got it. I try to never do that in my code because it might be complicated to understand for some other people and I think there are better patterns.

Reply:
I usually never use static variables myself, but this seemed reasonable here (where I first suggested this, I didn't realize the method was static)